httpserver/src/main/kotlin/tbdex/sdk/httpserver/handlers/SubmitOrder.kt

```diff
@@ -49,6 +49,12 @@ suspend fun submitOrder(
     return
   }
 
+  if(message.metadata.protocol != exchange.first().metadata.protocol) {
```

Comment:
Do we want to handle the case of supporting multiple protocols?

Reply:
Not within an exchange. I think a PFI can support multiple protocols concurrently, but each exchange should be consistent with one version.

Reply:
then should the protocol be specified once at the start of the exchange instead of in each message?

Reply:
We consider version to be part of the verification phase of each message because the protocol makes no assertions with respect to session requirements for a given exchange; AKA one could imagine an upgrade happening mid-exchange in which case we choose to error-out to prevent unforeseen consequences and include relevant diagnostic information

I'm open to feedback or improvement proposals here

Reply:
@decentralgabe yeah it is a bit wonky in that way. Right now, we don't have an explicit "exchange" data model at the protocol layer. 

To create an exchange, there is a `POST /exchanges` call, which contains an RFQ + reply URL. I guess we could introduce version as an argument here, but that means it's not part of the core `protocol` and layered in by the http API, which seems incorrect. 

+1 to @KendallWeihe 's comment about upgrades happening mid exchange (likely in any PFI with non-trivial traffic),  but yeah if you can think of any nicer ways to do this, open to improvements!

Reply:
makes sense to me as is 😄